docs/src/main/sphinx/object-storage/file-system-s3.md

```diff
@@ -0,0 +1,90 @@
+# S3 file system support
+
+Trino includes a native implementation to access [Amazon
+S3](https://aws.amazon.com/s3/) and compatible storage systems with a catalog
+using the Delta Lake, Hive, Hudi, or Iceberg connectors.
+
+Enable the native implementation with `fs.native-s3.enabled=true` in your
+catalog properties file.
+
+## General configuration
+
+Use the following properties to configure general aspects of S3 file system
+support:
+
+:::{list-table}
+:widths: 40, 60
+:header-rows: 1
+
+* - Property
+  - Description
+* - `fs.native-s3.enabled`
+  - Activate the native implementation for S3 storage support, and deactivate
+    all [legacy support](file-system-legacy). Defaults to `false`. Must be set
+    to `true` for all other properties be used.
+* - `s3.endpoint`
+  - Required endpoint URL for S3.
+* - `s3.region`
+  - Required region name for S3.
+* - `s3.path-style-access`
+  - Use path-style access for all requests to S3
+* - `s3.sse.type`
+  - Set the type of S3 server-side encryption (SSE) to use. Defaults to `NONE` for no
```

Comment:
AWS S3 actually does automatic SSE when this is set to `NONE`, but S3-compatible storage might behave differently. So for AWS, there should be no difference between `NONE` and `S3`.

Reply:
Done